src/serial/private/utils/list_serialports_windows.nim

```diff
@@ -44,10 +44,30 @@ proc SetupDiDestroyDeviceInfoList(DeviceInfoSet: HDEVINFO): bool
 
 iterator listSerialPorts*(): string =
   ## Iterates through a list of serial port names currently available on the system.
-  ##
-  ## This is based upon the `CEnumerateSerial::QueryUsingSetupAPI` method from `CEnumerateSerial`: http://www.naughter.com/enumser.html
 
-  # First create a "device info set" for the device interface GUID
+  
+  ## First we enumerate the USB serial ports from the appropritate registry node, 
+  ## this is because some of them are reported not to show up using the second method below.
+  ## see https://github.com/euantorano/serial.nim/issues/26
+  var usbSerialPortList = newSeq[string]()
```

Comment:
Might be slightly faster using `sets.HashSet` rather than a seq, though I haven't tested:

```suggestion
import sets
  
var usbSerialPortList = initHashSet[string]()
```

Then later on the `sets.containsOrIncl()` proc can be used to check and add an entry with a single call.

Reply:
I've added a commit which uses HashSet.  Faster or not, it makes more semantic sense.  I'll ask LucaWolf if the devices in question show up in HKLM\HARDWARE\DEVICEMAP\SERIALCOMM.  If they do, I agree that it would be much more elegant to replace the whole procedure with one that enumerates that key set.